addons/mail/static/src/core/common/thread.js

```diff
@@ -200,6 +200,12 @@ export class Thread extends Component {
             if (!this.env.chatter || this.env.chatter?.fetchMessages) {
                 if (this.env.chatter) {
                     this.env.chatter.fetchMessages = false;
+                    const needactionMessages = this.props.thread.messages.filter(
+                        (msg) => msg.needaction && msg.message_type !== "user_notification"
+                    );
+                    if (needactionMessages.length) {
+                        this.props.thread.markMessagesAsRead(needactionMessages.map((m) => m.id));
+                    }
```

Comment:
```suggestion
                    const needactionMessageIds = this.props.thread.messages.filter(
                        (msg) => msg.needaction && msg.message_type !== "user_notification"
                    ).map(msg => msg.id);
                    if (needactionMessageIds.length) {
                        this.props.thread.markMessagesAsRead(needactionMessageIds);
                    }
```
readable and simple indicating we care about the message ids